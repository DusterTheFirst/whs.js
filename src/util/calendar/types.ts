/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

/** The information about a calendar */
export interface ICalendarInformation {
    /** The school days */
    schoolDays: ICalendarSchoolDay[];
    /** When the calendar was updated */
    updated: number;
}

/** A School Day */
export interface ICalendarSchoolDay {
    /** The day number */
    dayNumber: SchoolDay;
    /** If the school day is a half day */
    isHalf: boolean;
    /** Any meta information regarding the day */
    meta?: string;
    /** The date the day is on formatted as YYYY-MM-DD */
    date: string;
}

/** A numeric value denoting the school day in the 7 day cycle */
export enum SchoolDay {
    One = 1,
    Two = 2,
    Three = 3,
    Four = 4,
    Five = 5,
    Six = 6,
    Seven = 7
}

export const allSchoolDays: SchoolDay[] = [
    SchoolDay.One,
    SchoolDay.Two,
    SchoolDay.Three,
    SchoolDay.Four,
    SchoolDay.Five,
    SchoolDay.Six,
    SchoolDay.Seven
];