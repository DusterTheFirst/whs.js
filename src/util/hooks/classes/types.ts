/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import { ClassType, IClassType } from "../../class/type";

export interface IUseClass<C extends IClassType<ClassType>> {
    savedValue?: C;
    tempValue: C;
    updated: boolean;
    save(): void;
    update(data: Partial<C>): void;
}