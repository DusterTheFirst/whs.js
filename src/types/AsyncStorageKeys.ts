enum AsyncStorageKey {
    Calendar = "@whs/calendar",
    Blocks = "@whs/blocks"
}

export default AsyncStorageKey;