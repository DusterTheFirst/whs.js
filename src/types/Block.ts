/*!
 * Copyright (C) 2018  Zachary Kohnen (DusterTheFirst)
 */

/** A class block */
export interface IClassBlock<T extends BlockColor = BlockColor> {
    /** The name of the block */
    name: string;
    /** The room the class is in */
    room: number;
    /** The teacher for the class */
    teacher: string;
    /** The class color */
    color: T;
    /** The days the class block should be on */
    days: SchoolDay[];
    /** If the class block has a lab */
    hasLab: boolean;
}

export type ColorBlockMap = {
    [T in BlockColor]: IClassBlock<T>
};

/** An avisory block */
export interface IAdvisory {
    /** The name of the block */
    name: "Advisory";
    /** The room number of the block */
    room: number;
    /** The teacher for the block */
    teacher: string;
}

/** The block numbers */
export enum Block {
    First,
    Second,
    Advisory,
    Third,
    Fourth,
    Fifth,
    Sixth
}

// TODO: hex values?
/** Colors of the blocks */
export enum BlockColor {
    Red = "red",
    Orange = "orange",
    Yellow = "gold",
    Green = "green",
    Blue = "blue",
    Purple = "purple",
    Tan = "tan",
    None = "none"
}

export enum BlockColorColors {
    Red = "#FF8080",
    Orange = "orange",
    Yellow = "gold",
    Green = "green",
    Blue = "blue",
    Purple = "purple",
    Tan = "tan",
    None = "none"
}

/** The lunch blocks */
export enum LunchBlock {
    First,
    Second,
    Third
}

/** The school days */
export enum SchoolDay {
    Day1 = 1,
    Day2,
    Day3,
    Day4,
    Day5,
    Day6,
    Day7
}

export const AllDays: SchoolDay[] = [
    SchoolDay.Day1,
    SchoolDay.Day2,
    SchoolDay.Day3,
    SchoolDay.Day4,
    SchoolDay.Day5,
    SchoolDay.Day6,
    SchoolDay.Day7
];

// /** Check if the display block given is a free block */
// export function isFree(block: IBlock): block is IFreeBlock {
//     return block.name === "Free";
// }

// /** Check if the display block given is an advisory block */
// export function isAdvisory(block: IBlock): block is IAdvisory {
//     return block.name === "Advisory";
// }

// /** Check if the display block given is a class block */
// export function isClassBlock(block: IBlock): block is IClassBlock {
//     return !isFree(block) && !isAdvisory(block);
// }

// /** Check if the display block given is a lab block */
// export function isLabBlock(block: IBlock): block is ILabBlock {
//     return isClassBlock(block) && (block as ILabBlock).isLab === true;
// }

// export type IBlock = IFreeBlock | IAdvisory | IClassBlock;