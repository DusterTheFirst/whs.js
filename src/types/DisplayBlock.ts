/** A class block that is prepared to be displayed on screen */
export interface IDisplayClassBlock {
    /** The name of the block */
    name: string;
    /** The room the class is in */
    room: number;
    /** The teacher for the class */
    teacher: string;
    /** The class color */
    color?: BlockColor;
    /** The block number */
    blockNumber: Block;
    /** The Lunch block  */
    lunchBlock?: LunchBlock;
}

/** A class block that is prepared to be displayed on the screen that has a lunch block */
export interface IDisplayLunchBlock extends IDisplayClassBlock {
    /** The Lunch block */
    lunchBlock: LunchBlock;
}

/** An avisory block that is prepared to be displayed on the screen */
export interface IDisplayAdvisory {
    /** The name of the block */
    name: "Advisory";
    /** The room number of the block */
    room: number;
    /** The teacher for the block */
    teacher: string;
    /** The number for the block */
    blockNumber: Block.Advisory;
}

/** A free block that is prepared to be displayed on the screen */
export interface IDisplayFree {
    /** The name of the block */
    name: "Free";
    /** The number of the block */
    blockNumber: Block;
    /** THe color of the block (if not a first period) */
    color?: BlockColor;
}

/** The block numbers */
export enum Block {
    First,
    Second,
    Advisory,
    Third,
    Fourth,
    Fifth,
    Sixth
}

// TODO: Fixed hex values?
export enum BlockColor {
    Red = "red",
    Orange = "orange",
    Yellow = "gold",
    Green = "green",
    Blue = "blue",
    Purple = "purple",
    Tan = "tan"
}

/** The lunch blocks */
export enum LunchBlock {
    First,
    Second,
    Third
}

/** Check if the display block given is a free block */
export function isFree(block: IDisplayBlock): block is IDisplayFree {
    return block.name === "Free";
}

/** Check if the display block given is an advisory block */
export function isAdvisory(block: IDisplayBlock): block is IDisplayAdvisory {
    return block.blockNumber === Block.Advisory;
}

/** Check if the display block given is a class block */
export function isClassBlock(block: IDisplayBlock): block is IDisplayClassBlock {
    return !isFree(block) && !isAdvisory(block);
}

/** Check if the display block given is a lunch class block */
export function isLunchBlock(block: IDisplayBlock): block is IDisplayLunchBlock {
    return isClassBlock(block) && block.lunchBlock !== undefined;
}

export type IDisplayBlock = IDisplayFree | IDisplayAdvisory | IDisplayClassBlock | IDisplayLunchBlock;