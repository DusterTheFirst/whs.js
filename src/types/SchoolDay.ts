import { LunchBlock } from "./Block";

export interface ISchoolDay {
    dayNumber: ValidSchoolDayNumber;
    lunch?: LunchBlock;
    isHalf: boolean;
    meta?: string;
    date: Date;
}

export type ValidSchoolDayNumber = 0 | 1 | 2 | 3 | 4 | 5 | 6;