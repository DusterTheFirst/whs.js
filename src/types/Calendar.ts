import { ISchoolDay } from "./SchoolDay";

export interface ICalendarInformation {
    /** The school days */
    schoolDays: ISchoolDay[];
    /** Events that take place */
    events: IEvent[];
}

/** An event */
export interface IEvent {
    /** The start time of the event */
    start: Date;
    /** The end time of the event */
    end: Date;
    /** The timestamp of the event */
    stamp: Date;
    /** The name of the event */
    name: string;
    /** The description of the event */
    description?: string;
    /** The location of the event */
    location?: string;
    /** If the event is all day */
    isAllDay: boolean;
}