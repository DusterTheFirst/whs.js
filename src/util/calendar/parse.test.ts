/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import { readFileSync } from "fs";
import { resolve } from "path";
import parseCalendar from "./parse";
import { SchoolDay } from "./types";

/** The mock calendar data */
const calendarData = readFileSync(resolve(__dirname, "../../__mocks__/calendar.ics")).toString();

describe("Tests parser with mock data", () => {
    const calendar = parseCalendar(calendarData);

    it("Parses schooldays and matches the snapshot", () => {
        expect(calendar.schoolDays).toMatchSnapshot();
    });

    it("All school days have valid SchoolDay enum values", () => {
        // tslint:disable-next-line: strict-type-predicates
        expect(calendar.schoolDays.every(x => SchoolDay[x.dayNumber] !== undefined)).toBeTruthy();
    });

    it("Sets updated to a date", () => {
        expect(calendar.updated).toBeTruthy();
    });
});