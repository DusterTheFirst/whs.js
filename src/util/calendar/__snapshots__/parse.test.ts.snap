// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests parser with mock data Parses schooldays and matches the snapshot 1`] = `
Array [
  Object {
    "date": "2020-02-05",
    "dayNumber": 7,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2019-11-06",
    "dayNumber": 6,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2020-01-08",
    "dayNumber": 4,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2020-05-06",
    "dayNumber": 4,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2020-04-01",
    "dayNumber": 3,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2020-06-05",
    "dayNumber": 2,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2019-12-04",
    "dayNumber": 2,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2019-11-27",
    "dayNumber": 1,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2019-10-16",
    "dayNumber": 1,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2020-03-04",
    "dayNumber": 1,
    "isHalf": true,
    "meta": "",
  },
  Object {
    "date": "2019-10-15",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-16",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-05",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-25",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-25",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-13",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-03",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-20",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-11",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-30",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-18",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-08",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-04",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-28",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-06",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-15",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-17",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-28",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-07",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-09",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-21",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-02",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-03",
    "dayNumber": 7,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-06",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-15",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-24",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-07",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-10",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-12",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-27",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-11",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-02",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-27",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-24",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-05",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-27",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-15",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-14",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-26",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-08",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-01",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-04",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-19",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-01",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-18",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-16",
    "dayNumber": 6,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-14",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-09",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-07",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-15",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-01",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-23",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-11",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-13",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-12",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-04",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-18",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-14",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-13",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-03",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-26",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-29",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-17",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-23",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-25",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-20",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-05",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-10",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-27",
    "dayNumber": 5,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-03",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-10",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-16",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-23",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-17",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-22",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-28",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-13",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-08",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-04",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-12",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-22",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-10",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-13",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-31",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-26",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-06",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-22",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-22",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-19",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-02",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-14",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-06",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-11",
    "dayNumber": 4,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-11",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-07",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-18",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-19",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-28",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-16",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-12",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-01",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-15",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-20",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-09",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-03",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-25",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-21",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-31",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-05",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-21",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-13",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-21",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-30",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-27",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-09",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-10",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-12",
    "dayNumber": 3,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-26",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-18",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-06",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-20",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-29",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-10",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-24",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-04",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-18",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-17",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-27",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-19",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-30",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-09",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-30",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-02",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-04",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-13",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-12",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-08",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-14",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-07",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-17",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-11",
    "dayNumber": 2,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-03",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-11",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-05-22",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-16",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-08-28",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-13",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-04-29",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-07",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-26",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-29",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-11-19",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-06",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-08",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-17",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-31",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-23",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-08-30",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-12",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-01-16",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-09",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-02-10",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-12-05",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-09-03",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-06-17",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2020-03-18",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-10-28",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
  Object {
    "date": "2019-08-29",
    "dayNumber": 1,
    "isHalf": false,
    "meta": "",
  },
]
`;
